test/functional/admin/attachments_controller_test.rb

```diff
@@ -250,7 +250,10 @@ def self.supported_attachable_types
   end
 
   test "POST :create with an edition belonging to a locked document does not save the attachment" do
-    edition = create(:edition, document: build(:document, locked: true))
+    document = build(:document)
+    edition = create(:unpublished_edition, document: document)
```

Comment:
I'd suggest it'd be good to pass in a locked one so we could avoid the need to modify the document. For example: `create(:unpublished_edition, document: build(:document, locked: true))` - but do agree it should be avoided if there are no args.

Reply:
My rationale for this is a hiccup with the locked logic. the original create line now throws a locked document error, as the document is created first, saved, then applied to the edition. The edition lock check then fires on the save step of the edition creation.

I also think this is a bad way to do things, but for the purposes of the test it may be the only way. I'll try the one-line approach again to see if there is a workaround.